core/src/main/java/org/apache/druid/math/expr/Expr.java

```diff
@@ -494,6 +496,18 @@ public String toString()
   {
     return NULL_LITERAL;
   }
+
+  @Override
+  public int hashCode()
+  {
+    return getClass().hashCode();
```

Comment:
A reasonable default for Expr could be to use the stringified form for hashCode and equals:

```
  /**
   * Convert the {@link Expr} back into parseable string that when parsed with
   * {@link Parser#parse(String, ExprMacroTable)} will produce an equivalent {@link Expr}.
   */
  String stringify();
```

Reply:
Removed this. Each implementation will be responsible for defining their equals and hashcode